src/main/groovy/com/tmiyamon/config/ConfigPlugin.groovy

```diff
@@ -1,31 +1,58 @@
 package com.tmiyamon.config
 
+import com.android.build.gradle.LibraryExtension
 import com.tmiyamon.config.task.GenerateSettingsTask
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import com.android.build.gradle.api.BaseVariant
 import com.android.build.gradle.AppExtension
+import org.gradle.api.Task
 
 class ConfigPlugin implements Plugin<Project> {
+
+    enum ProjectType {
+
+        ANDROID_APPLICATION('com.android.application', AppExtension),
+        ANDROID_LIBRARY('com.android.library', LibraryExtension)
+
+        final String pluginId
+        final Class extensionType
+
+        ProjectType(String pluginId, Class extensionType) {
+            this.pluginId = pluginId
+            this.extensionType = extensionType
+        }
+    }
+
     @Override
     void apply(Project project) {
-        project.plugins.withId('com.android.application') {
-            def android = project.extensions.getByType(AppExtension)
-            android.applicationVariants.all { BaseVariant variant ->
-
-                def task = project.tasks.create(
-                    name: "generate${variant.name.capitalize()}Settings",
-                    type: GenerateSettingsTask) {
-                        packageName    variant.generateBuildConfig.buildConfigPackageName
-                        flavorName     variant.flavorName
-                        buildTypeName  variant.buildType.name
-                        variantDirName variant.dirName
-                    }
+        ProjectType.values().each {
+            createCodeGenerationTasks(project, it)
```

Comment:
The only thing I'm thinking about is that this method may or may not create the task, depending on the nature of the project? Maybe there is a smart way to split this functions or change the method name.